src/RefractDataStructure.cc

```diff
@@ -25,14 +25,26 @@ namespace drafter {
     struct ElementData {
         typedef T ElementType;
 
-        typedef typename T::ValueType ValueType;
+        typedef typename ElementType::ValueType ValueType;
         typedef snowcrash::SourceMap<ValueType> ValueSourceMapType;
 
-        typedef std::tuple<ValueType, ValueSourceMapType, bool> ValueInfo; // [value, sourceMap, validity(by LiteralTo<>)]
+        static const bool IsPrimitive = std::conditional<
```

Comment:
Consider whether using a `static bool` here is what we want. The STL comes with the approach of using free type constructors like the following for such use cases:
```C++
template<typename T>
struct is_primitive {
  using type = std::integral_constant<bool, !std::is_same<T, refract::RefractElements>::value>
};
```
Keep in mind that std::true_type and std::false_type implicitly convert to a runtime bool if necessary, but by itself it is a compile-time concept that doesn't have a runtime imprint. The static bool above does.